megatron/training/training.py

```diff
@@ -1046,6 +1049,25 @@ def training_log(loss_dict, total_loss_dict, learning_rate, decoupled_learning_r
         throughput = num_floating_point_operations(args, batch_size) / (
             elapsed_time_per_iteration * 10**12 * args.world_size)
 
+        # Calculate tokens per second
+        tokens_per_iteration = args.global_batch_size * args.seq_length
+        tokens_per_sec = tokens_per_iteration / elapsed_time_per_iteration
```

Comment:
```suggestion
        tokens_per_sec = tokens_per_iteration / elapsed_time_per_iteration / torch.distributed.get_world_size()
```
IMO It makes much more sense to divide the throughput with the number of devices, this way we can compare across configs and methods. It doesn't provide any relevant information having `124334` or `8878748676` apart from computing when the job is going to finish but for that we already have `eta`, right?

Reply:
made the change for tok/sec/gpu. the elapsed time per iter per rank is quite a bit more work, yeah, so maybe we can ignore that for now?

Reply:
yes, not a priority IMO

Reply:
Logging per gpu metrics (either the iteration time or loss per dp replica let's say) it's overkilled and will be a completely mess for the Wandb panel. 

Am I the only one who still thinks `tokens_per_sec ` is useless as we already have `tokens_per_sec_per_gpu `? Can we get rid of that?

Reply:
You decide. I'm strongly biased towards minimalism, might be my problem too 😅